MiniApp/Classes/MiniApp.swift

```diff
@@ -51,6 +51,26 @@ public class MiniApp: NSObject {
         return realMiniApp.createMiniApp(appId: appId, completionHandler: completionHandler, messageInterface: messageInterface)
     }
 
+    /// Cache the Custom permissions status for a given MiniApp ID
+    /// - Parameters:
+    ///   - appId: Mini AppId String value
+    ///   - permissionList: List of MASDKCustomPermissionModel class that contains the MiniAppCustomPermissionType and MiniAppCustomPermissionGrantedStatus
+    public func setCustomPermissions(forMiniApp appId: String, permissionList: [MASDKCustomPermissionModel]) {
+        if !appId.isEmpty {
+            return realMiniApp.storeCustomPermissions(forMiniApp: appId, permissionList: permissionList)
+        }
+    }
+
+    /// Get the list of supported Custom permissions and its status for a given Mini app ID
+    /// - Parameter appId: Mini AppId String value
+    /// - Returns: List of MASDKCustomPermissionModel that contains the details of every custom permission type, status and description.
+    public func getCustomPermissions(forMiniApp appId: String) -> [MASDKCustomPermissionModel]? {
```

Comment:
I think that the return value shouldn't be nullable. This should always return the full list of supported permissions. If they have never requested one of the permissions, then that permission should be returned as `DENIED`. This is so the Host App can have a consistent permissions screen for every Mini App. Also a user may wish to grant a particular permission to a Mini App which the Mini App hasn't requested yet. However, the downside is that it would be possible to grant a permission which the Mini App doesn't actually use.

Do you think that makes sense?

Reply:
Agreed 👍 , return shouldn't be nullable. Will make changes